Gems/Blast/Code/Source/Components/BlastSystemComponent.cpp

```diff
@@ -218,16 +218,18 @@ namespace Blast
 
         jobCompletion.StartAndWaitForCompletion();
 
-        // Run groups
-        for (auto i = 0; i < m_groups.size(); ++i)
+        // Remove empty groups
+        for (size_t i = 0; i < m_groups.size(); ++i)
         {
             if (m_groups[i].m_tkGroup->getActorCount() == 0)
             {
                 AZStd::swap(m_groups[i], m_groups.back());
                 m_groups.pop_back();
+                --i; // Decrease index so in the next loop we consider the same position (as another group has been put in)
             }
         }
```

Comment:
suggestion: Instead of the `--i;`, could we put the `++i;` in an `else` and remove from the for loop

```
for (size_t i = 0; i < m_groups.size();)
{
    if (m_groups[i].m_tkGroup->getActorCount() == 0)
    {
        AZStd::swap(m_groups[i], m_groups.back());
        m_groups.pop_back();
    }
    else
    {
        i++;
    }
}
```

Reply:
Yeah, good idea, I'll change it.

Reply:
Cool thanks!